packages/nimble-components/src/select/index.ts

```diff
@@ -49,51 +146,701 @@ export class Select extends FoundationSelect implements ErrorPattern {
+    }
+
+    /**
+     * Synchronize the `aria-disabled` property when the `disabled` property changes.
+     *
+     * @param prev - The previous disabled value
+     * @param next - The next disabled value
+     *
+     * @internal
+     */
+    public override disabledChanged(prev: boolean, next: boolean): void {
+        if (super.disabledChanged) {
+            super.disabledChanged(prev, next);
+        }
+        this.ariaDisabled = this.disabled ? 'true' : 'false';
+    }
+
+    /**
+     * Reset the element to its first selectable option when its parent form is reset.
+     *
+     * @internal
+     */
+    public override formResetCallback(): void {
+        this.setProxyOptions();
+        // Call the base class's implementation setDefaultSelectedOption instead of the select's
+        // override, in order to reset the selectedIndex without using the value property.
+        super.setDefaultSelectedOption();
+        if (this.selectedIndex === -1) {
+            this.selectedIndex = 0;
+        }
+    }
+
+    // Prevents parent classes from resetting selectedIndex to a positive
+    // value while filtering, which can result in a disabled option being
+    // selected.
+    protected override setSelectedOptions(): void {
+        if (this.open && this.selectedIndex === -1) {
+            return;
+        }
+
+        super.setSelectedOptions();
+    }
+
+    protected positionChanged(
+        _: SelectPosition | undefined,
+        next: SelectPosition | undefined
+    ): void {
+        this.positionAttribute = next;
+        this.setPositioning();
+    }
+
+    /**
+     * Updates the proxy's size property when the size attribute changes.
+     *
+     * @param prev - the previous size
+     * @param next - the current size
+     *
+     * @override
+     * @internal
+     */
+    protected override sizeChanged(
+        prev: number | undefined,
+        next: number
+    ): void {
+        super.sizeChanged(prev, next);
+
+        if (this.proxy) {
+            this.proxy.size = next;
+        }
+    }
+
+    protected openChanged(): void {
+        if (!this.collapsible) {
+            return;
+        }
+
+        if (this.open) {
+            this.initializeOpenState();
+            this.indexWhenOpened = this.selectedIndex;
+            if (this.filterMode === FilterMode.none) {
+                DOM.queueUpdate(() => this.focus());
+            }
+
+            return;
+        }
+
+        this.filter = '';
+        if (this.filterInputElement) {
+            this.filterInputElement.value = '';
+        }
+
+        this.ariaControls = '';
+        this.ariaExpanded = 'false';
+    }
+
+    /**
+     * Updates the selectedness of each option when the list of selected options changes.
+     *
+     * @param prev - the previous list of selected options
+     * @param next - the current list of selected options
+     *
+     * @override
+     * @internal
+     */
+    protected override selectedOptionsChanged(
```

Comment:
@jattasNI @rajsite, from here until `clearSelection` _except_ for `filterOptions` is copied directly from FAST.